libbeat/outputs/elasticsearch/client.go

```diff
@@ -82,6 +84,12 @@ func NewClient(
 		index:  index,
 		params: params,
 	}
+
+	client.Connection.onConnectCallback = func() {
```

Comment:
wrapping func once again is a little annoying. Why not have `type connectCallback func(*Client) error` and reuse type all over the place with `var noneCallback connectCallback = func(*Client) error { return nil}`. Set `noneCallback` on `&Client` and if parameter `!= nil` assign it.